ui/apps/dashboard/src/app/(dashboard)/env/[environmentSlug]/apps/[externalID]/ResyncModal.tsx

```diff
@@ -0,0 +1,97 @@
+import { useState } from 'react';
+import ArrowPathIcon from '@heroicons/react/20/solid/ArrowPathIcon';
+import { Button } from '@inngest/components/Button';
+import { toast } from 'sonner';
+
+import Modal from '@/components/Modal';
+import { DeployFailure } from '../../deploys/DeployFailure';
+import { deployViaUrl, type RegistrationFailure } from '../../deploys/utils';
+
+type Props = {
+  isOpen: boolean;
+  onClose: () => void;
+  url: string;
+};
+
+export default function ResyncModal({ isOpen, onClose, url }: Props) {
+  const [failure, setFailure] = useState<RegistrationFailure>();
+  const [isSyncing, setIsSyncing] = useState(false);
+
+  async function onSync() {
+    setIsSyncing(true);
+
+    let failure;
+    try {
+      // TODO: This component is using legacy syncs stuff that needs
+      // reorginization and/or refactoring. We should use a GraphQL mutation
+      // that gets the last sync URL, rather than relying on the UI to find it.
+      failure = await deployViaUrl(url);
+
+      setFailure(failure);
+      if (!failure) {
+        toast.success('Synced app');
+        onClose();
+      }
+    } catch {
+      setFailure({
+        errorCode: undefined,
+        headers: {},
+        statusCode: undefined,
+      });
+    } finally {
+      setIsSyncing(false);
+    }
+  }
+
+  return (
+    <Modal
```

Comment:
There is a shared modal in components with those styles, this Modal is an old one that we're removing slowly (uses headless ui rather than radix)